### PR TITLE
support fastify with swagger plugin

### PR DESCRIPTION
**Pull request by bbetts-godaddy**
## Summary

Add a Fastify hook to the `@gasket/plugin-swagger` package to allow for Fastify to be used in place of express. 

## Changelog

- create Fastify hook

## Test Plan

Screen Shot 2022-08-01 at 1 56 53 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -75,5 +75,8 @@
   "workspaces": [
     "./packages/*",
     "scripts/generate-docs-index"
-  ]
+  ],
+  "dependencies": {
+    "browserslist-config-godaddy": "^3.0.1"
```
Comment (jpage-godaddy) [inline package.json:80]: This is an internal package, no? Probably shouldn't be a dependency of this open source package.

**Review by jpage-godaddy — APPROVED** [review]

File: packages/gasket-plugin-swagger/lib/index.js [hunk 2]
```diff
@@ -108,6 +108,30 @@ module.exports = {
...
+      timing: {
+        before: ['@gasket/plugin-nextjs']
+      },
+      handler: async function fastify(gasket, app) {
+        const { swagger, root } = gasket.config;
+        const { ui = {}, apiDocsRoute, definitionFile } = swagger;
+
+        const swaggerSpec = await loadSwaggerSpec(root, definitionFile, gasket.logger);
+
+        app.register(require('@fastify/swagger'), {
```
Comment (agerard-godaddy) [inline packages/gasket-plugin-swagger/lib/index.js:128]: 👍 Can you also delay the `require('swagger-ui-express')` to within the `express` lifecycle handler? It doesn't need to be loaded if it's not used.

File: packages/gasket-plugin-swagger/package.json [hunk 3]
```diff
@@ -39,8 +39,10 @@
   },
   "homepage": "https://github.com/godaddy/gasket/tree/main/packages/gasket-plugin-swagger",
   "dependencies": {
+    "@fastify/swagger": "^6.0.0",
     "@types/swagger-ui-express": "^4.1.3",
     "express": "^4.16.3",
+    "fastify": "^4.1.0",
```
Comment (agerard-godaddy) [inline packages/gasket-plugin-swagger/package.json:45]: `fastify` seems to only be used in your tests, and thus should be in `devDependencies`. It seems odd that `express` is not a dev dependency also, could it be moved, too?

**Review by agerard-godaddy — COMMENTED** [review]
Looking great! Are there any configuration nuances between using this with express vs fastify that need to be called out in the Readme docs? Currently, it has a link to [swagger-ui-express](https://github.com/scottie1984/swagger-ui-express) options for the `ui` property. Are these the same? is there a centralized doc that would make more sense to link to?

File: packages/gasket-plugin-swagger/package.json (hunk 3 above)
Comment (bbetts-godaddy) [reply packages/gasket-plugin-swagger/package.json:45]: yeah, I don't think I see express used anywhere in the package

**Review by agerard-godaddy — APPROVED** [review]

**Pull request merged**